kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/manager/ResourceManager.java

```diff
@@ -43,6 +53,9 @@
 public class ResourceManager {
     private static final Logger LOGGER = LoggerFactory.getLogger(ResourceManager.class);
     private static ResourceManager instance;
+    private static String koDeploymentName = Constants.KO_DEPLOYMENT_NAME;
+    private static ExtensionContext testContext;
+    private static final Map<String, Stack<ResourceItem<?>>> storedResources = new LinkedHashMap<>();
```

Comment:
does the order of entries matter? I'm wondering if it could be a `ConcurrentHashMap` and you could replace the `synchronized` block later with:

```
storedResources.computeIfAbsent(getTestContext().getDisplayName(), k -> new Stack<>())
    .push(new ResourceItem<>(() -> deleteResource(resource), resource));
```